src/main/java/teammates/storage/entity/AdminEmail.java

```diff
@@ -88,7 +87,7 @@ public void setSendDate(Date sendDate) {
     }
 
     public String getEmailId() {
-        return Key.create(AdminEmail.class, this.emailId).toWebSafeString();
+        return emailId == null ? null : Key.create(AdminEmail.class, this.emailId).toWebSafeString();
```

Comment:
When is this null?

Reply:
Many instances in production code (not just test code, otherwise I'll also complain) don't require this attribute to be equipped with `emailId`. So it is already optional to begin with.

Use case: if you want to pass it to functions that don't require the existence of `emailId`.

It may not be the best deciding factor on what to count a field as "optional" or not, but if you really want to pick on that, I'll suggest you relook at all the attributes builders instead of just picking on this one.